### ISSUE TITLE
Portrait mode on Android

### ISSUE THREAD

**Issue by Marus** [wildabeast/BarcodeScanner#70]
Hi,

I''m just wondering if it is possible to use the barcode scanning plugin in portrait mode on Android? Seems like a strange restriction to have it limited to landscape when the majority of UIs for phone applications are portrait primarily, and from a user experience point of view it isn't very nice to have to constantly switch between portrait and landscape.

**Comment by nickrouty** [comment]
Seems like a shame that a new plugin was started to address this -- why not submit a pull request? Ionic uses this version within the ngCordova setup. Be nice to just have this plugin 'fixed' for android.

**Comment by Dbuggerx** [comment]
@nickrouty My apologies...I've been busy with other personal projects since I implemented the "portrait mode". 
Is it still relevant to submit a pull request to this repository? (I guess so, since you've pointed out that ngCordova still uses it) If so, I guess I can find some time to update my code, rebase my changes and submit a PR (maybe even submit it to phonegap).

**Comment by ycanales** [comment]
Found [this fork](https://github.com/trongrg/phonegap-plugin-barcodescanner), it's recent and forked from the new official repo, it allows for portrait and landscape and the area is larger too.

Tested it and works fine in Android / Nexus 5.

**Comment by nickrouty** [comment]
@Dbuggerx No apologies needed. Not sure if @wildabeast is continuing to maintain the project or not - if so, a pull request would be good as I'm sure most people find this project first before finding updated versions. Yours worked great, thanks for your work!

**Comment by Dr-Steve** [comment]
+1 for this, taking a look in to the fork in the meantime.

**Comment by Duovili** [comment]
have you guys got this to work? I tried changing it myself and it made soo many problems

**Comment by Dr-Steve** [comment]
@Duovili I got Dbuggerx's fork to work at https://github.com/Dbuggerx/BarcodeScanner

I am using angular with cordova.  I first had to remove my platforms so that changes in the plugin directory would propagate down to the devices:

cordova platforms remove android

I then removed the old plugin that was installed by doing a 

```
cd plugins
rm -rf com.phonegap.plugins.barcodescanner
```

And then cloned his fork into the plugin directory with

```
git clone https://github.com/Dbuggerx/BarcodeScanner.git
```

I then renamed the clone dir to match the old plugin:

```
mv BarcodeScanner com.phonegap.plugins.barcodescanner
```

I think that I also had to install it from that directory:

```
cordova plugin install ./com.phonegap.plugins.barcodescanner
```

I then re-added all of my platforms, and it worked.  Note that your process may be a bit different, depending on your directory structure and whatnot.  Some people [here](http://forum.ionicframework.com/t/cordova-plugin-not-updating/4422) clain that you can get by without removing your platforms and re-adding them using the command ```cordova prepare android ```, but that did not work for me.  Also note that your config may be different depending on how you had originally installed your plugin.  modifying fetch.json to have something like this may help:

```
    "com.phonegap.plugins.barcodescanner": {
        "source": {
            "type": "local",
            "path": "https://github.com/Dbuggerx/BarcodeScanner.git"
        }

```

Also you may find it easier to remove your old plugin then install using the link:

```
cordova plugin install https://github.com/Dbuggerx/BarcodeScanner.git
```

**Comment by Duovili** [comment]
but is it by default set to portrait?

**Comment by Dr-Steve** [comment]
Yes, if you get that fork working then it will open in portrait mode.

**Comment by Duovili** [comment]
Thanks this made my life soo much easier on an app I am bulding

**Comment by Dr-Steve** [comment]
@Dbuggerx is the real hero here.

**Comment by Duovili** [comment]
@Dbuggerx get that man a bells!

**Comment by vcajes** [comment]
+1 I want this!
